src/app.module.ts

```diff
@@ -15,16 +13,33 @@ import awsConfig from 'src/config/aws.config';
 import { PushNotificationAbstractModule } from './push-notification/abstract/push-notification-abstract.module.ts';
 import { ExpoAdapterModule } from './push-notification/expo-adapter/expo-adapter.module';
 import expoConfig from './config/expo.config';
+import { EmailAbstractModule } from './email/abstract/email-abstract.module';
+import emailConfig from './config/email.config';
+import { ResendAdapterModule } from './email/resend-adapter/resend-adapter.module';
+import { TemplateModule } from './templating/template.module';
+import { PugAdapterModule } from './templating/pug-adapter/pug-adapter.module';
 
 @Module({
   imports: [
     ConfigModule,
     AuthModule,
     MiddlewareModule,
-    TemplateModule,
-    EmailModule,
     SpaceshipModule,
     PrismaModule,
+    TemplateModule.forRoot({
```

Comment:
Just an idea I had just now, but we cooooould abstract this a bit more, like this:

```
TemplateModule.forRoot({
   adapter: TEMPLATE_ADAPTERS.PUG,
   config: {},
   isGlobal: true,
})
```

As an alternative. I like the other approach too because it's explicit in the adapter being used, and makes changing it clear, while the abstraction I'm proposing above is a bit more opaque. Just an idea!